src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -217,16 +239,27 @@ public Insets getBorderInsets(Component c, Insets newInsets) {
      */
     @SuppressWarnings("serial") // Superclass is not serializable across versions
     public static class InternalFrameBorder extends AbstractBorder implements UIResource {
-        private static final int corner = 14;
+        private static final int CORNER = 14;
 
         /**
          * Constructs a {@code InternalFrameBorder}.
          */
         public InternalFrameBorder() {}
 
-        public void paintBorder(Component c, Graphics g, int x, int y,
-                          int w, int h) {
+        /**
+         * Rounds a double to the nearest integer. It rounds 0.5 down,
+         * for example 1.5 is rounded to 1.0.
+         *
+         * @param d number to be rounded
+         * @return the rounded value
+         */
+        private static int roundHalfDown(double d) {
```

Comment:
The sun.java2d.pipe.Region#clipRound() seems have the same purpose, but it also clip the double to int.

Reply:
> The sun.java2d.pipe.Region#clipRound() seems have the same purpose, but it also clip the double to int.

Thank you, @mrserb. It's better to re-use the existing implementation. I tested with it, it gives the same result.

Reply:
@mrserb Thank you for the suggestion.